src/components/StakingPosition/StakingPosition.tsx

```diff
@@ -0,0 +1,107 @@
+import { BigNumber, ethers } from "ethers"
+import React from "react"
+import useSherlock from "../../hooks/useSherlock"
+import useWaitTx from "../../hooks/useWaitTx"
+import { PERIODS_IN_SECONDS } from "../../pages/Staking"
+import { convertSecondsToDurationString } from "../../utils/time"
+import { Button } from "../Button/Button"
+import styles from "./StakingPosition.module.scss"
+
+interface Props {
+  /**
+   * Position ID
+   */
+  id: number
+}
+
+const StakingPosition: React.FC<Props> = ({ id }) => {
+  const [lockedForSeconds, setLockedForSeconds] = React.useState<number>()
+  const [usdcBalance, setUsdcBalance] = React.useState<BigNumber>()
+  const [sherRewards, setSherRewards] = React.useState<BigNumber>()
+  const [stakingPeriod, setStakingPeriod] = React.useState<number>()
+  const isUnlocked = lockedForSeconds && lockedForSeconds <= 0
+
+  const { getPositionLockupEnd, getPositionSherRewards, getPositionUsdcBalance, unstake, restake } = useSherlock()
+  const { waitForTx } = useWaitTx()
+
+  /**
+   * Fetch the timestamp at which the position
+   * can be unstaked or restaked.
+   */
+  const handleFetchLockupPeriod = React.useCallback(async () => {
+    const lockupEnds = await getPositionLockupEnd(id)
+
+    const timeLeft = lockupEnds.sub(BigNumber.from((new Date().valueOf() / 1000).toFixed(0))).toNumber()
+    setLockedForSeconds(timeLeft)
+  }, [getPositionLockupEnd, id])
+
+  /**
+   * Fetch position's current USDC balance, claimable at the end
+   * of the lockup period.
+   */
+  const handleFetchUsdcBalance = React.useCallback(async () => {
+    const positionUsdcBalance = await getPositionUsdcBalance(id)
```

Comment:
Are you planning on using the GraphQL endpoint in a future PR? I guess this will work just fine, the indexer will just make things much faster.

Reply:
I merged master into this PR to get the GraphQL query and will migrate to using that as the source of truth. I have been using these for development.